internal/controllers/imagerepository_controller.go

```diff
@@ -516,8 +519,44 @@ func (r *ImageRepositoryReconciler) scan(ctx context.Context, obj *imagev1.Image
 		return 0, err
 	}
 
+	storedTags := make([]database.Tag, 0, len(filteredTags))
+
+	resCh := make(chan database.Tag, len(filteredTags))
+	var wg sync.WaitGroup
+	for _, tag := range filteredTags {
+		wg.Add(1)
+		go func(tag string, ch chan database.Tag) {
+			defer wg.Done()
+			res := database.Tag{
+				Name: tag,
+			}
+
+			if r.FetchDigests {
+				tagRef, err := name.ParseReference(strings.Join([]string{ref.Context().Name(), tag}, ":"))
+				if err != nil {
+					return
+				}
+				desc, err := remote.Head(tagRef, remote.WithContext(ctx))
+				if err != nil {
+					return
```

Comment:
What happens on error? Do we hide rate limits errors and other network issue? Wouldn't this make Flux rollback images to older versions and break the app deployments?